Controller/APIController.php

```diff
@@ -286,6 +287,14 @@ protected function validateHeaders(array $headers, $required = true)
             if (null === $headerValue && $required) {
                 throw new BadRequestHttpException(sprintf('"%s" header is required to use this method', $headerName));
             }
+
+            if (!empty($headerValidations[$headerName])) {
```

Comment:
The way its done here, the execution happens if there is no headerValue present, meaning that if its an optionalHeader but not sent, it will evaluate and validate assertion anyway. Maybe check if $headervalue is not null as in line 287 is a good approach.

What do you think?

Reply:
Yup, that's true. Fixed